packages/@react-aria/table/src/useTable.ts

```diff
@@ -109,7 +111,48 @@ export function useTable<T>(props: TableProps<T>, state: TableState<T>, ref: Ref
     return sortDirection && column ? formatMessage(`${sortDirection}Sort`, {columnName}) : undefined;
   }, [sortDirection, column, state.collection.columns]);
 
-  let descriptionProps = useDescription(sortDescription);
+
+  // do we have something that might be better suited for this? issue is that useInteractionModality
+  // doesn't return quite enough information
+  let [interactionType, setInteractionType] = useState('');
+  useEffect(() => {
+    let onPointerDown = (e) => {
+      setInteractionType(e.pointerType);
+    };
+    let onKeyDown = () => {
+      setInteractionType('keyboard');
+    };
+    let tableRef = ref.current;
+    tableRef.addEventListener('pointerdown', onPointerDown, true);
+    tableRef.addEventListener('pointerup', onPointerDown, true);
+    tableRef.addEventListener('keydown', onKeyDown, true);
+    tableRef.addEventListener('keyup', onKeyDown, true);
+    return () => {
+      tableRef.removeEventListener('pointerdown', onPointerDown, true);
+      tableRef.removeEventListener('pointerup', onPointerDown, true);
+      tableRef.removeEventListener('keydown', onKeyDown, true);
+      tableRef.removeEventListener('keyup', onKeyDown, true);
+    };
+  });
+  let interactionDescription = useMemo(() => {
+    let selectionMode = state.selectionManager.selectionMode;
+    let selectionBehavior = state.selectionManager.selectionBehavior;
+    // if we're in replace but can select multiple, then when using touch it's long press to enter selection mode
+    // if we can't tell what mode we're in, then we're probably in voice over, in which case we need the description to be set
+    // before the user interacts so they know they can enter selection mode before they actually have to interact
+    let message = formatMessage('longPressToSelect');
+    if (interactionType === 'touch') {
+      message = formatMessage('longPressToSelect');
+    }
+    return selectionBehavior === 'replace' && selectionMode !== 'none' && props.onAction ? message : undefined;
+  }, [state.selectionManager.selectionMode, state.selectionManager.selectionBehavior, formatMessage, interactionType]);
+
+  let sortDescriptionProps = useDescription(sortDescription);
+  let longPressDescriptionProps = useDescription(interactionDescription);
+  // todo: should mergeprops append describedby's?
```

Comment:
It looks like you switched form mergeprops to mergedescription, can this comment be deleted?

Reply:
i made up mergedescription, but don't know if i like it, since it only merges descriptions but currently accepts objects with aria-label/labelledby/etc

Reply:
Now that I think about it, could you not just append the strings and call `useDescription` once?